### ISSUE TITLE
Task: implement missing unit tests for new delegations repository functions

### ISSUE THREAD

**Issue by MVrachev** [repository-service-tuf/repository-service-tuf-worker#599]
### What is the task about?

We need better coverage of our repository functions introduced in:
https://github.com/repository-service-tuf/repository-service-tuf-worker/pull/593

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Sandipmandal25** [comment]
> I see that the tests for the functions introduced in this PR were created in this commit: [57b6c05](https://github.com/repository-service-tuf/repository-service-tuf-worker/commit/57b6c05bd78636e93b413c130cc420529d3fbc75) :
> 
> * read_role_deactivated_by_rolename: https://github.com/repository-service-tuf/repository-service-tuf-worker/blob/main/tests/unit/tuf_repository_service_worker/models/targets/test_crud.py#L549
> * update_role_to_deactivated: https://github.com/repository-service-tuf/repository-service-tuf-worker/blob/main/tests/unit/tuf_repository_service_worker/models/targets/test_crud.py#L576
> 
> Is there anything else to do or can this issue be closed [@MVrachev](https://github.com/MVrachev) ?


hi @camilaavilarinho @MVrachev wanted to give some context here 
so the public `metadata_delegation` function has tests, but `_add_metadata_delegation`, `_delete_metadata_delegation`, and `_update_metadata_delegation` are mocked out with `pretend.call_recorder` in those cases so their logic is not actually exercised.

I think that is the remaining coverage gap here. Would you be okay with me picking this up and adding direct unit tests for those private helpers?

**Comment by MVrachev** [comment]
It's normal that some helper functions will be mocked.
In those test files, we use unit tests where we mock the tests and that's okay.
The idea is that each function should be tested, and its main logic should not be mocked, only if there are new helper functions, they should be tested.

We additionally use functional tests to verify that the main functionality works correctly.

The idea of this issue is to add more unit tests for these functions.
You can see the functions with missing unit tests by running `tox -e test` and get:

<img width="1139" height="423" alt="Image" src="https://github.com/user-attachments/assets/9f8000e3-2cb4-4d1f-ba67-671f76c19537" />

I have a unit tests failing, so I will be grateful if you add unit tests for the missing functions and fix the failing unit tests.
Dou you want to be assigned on this @Sandipmandal25 ?

**Comment by Sandipmandal25** [comment]
> It's normal that some helper functions will be mocked.
> In those test files, we use unit tests where we mock the tests and that's okay.
> The idea is that each function should be tested, and its main logic should not be mocked, only if there are new helper functions, they should be tested.
> 
> We additionally use functional tests to verify that the main functionality works correctly.
> 
> The idea of this issue is to add more unit tests for these functions.
> You can see the functions with missing unit tests by running `tox -e test` and get:
> 
> <img width="1139" height="423" alt="Image" src="https://github.com/user-attachments/assets/9f8000e3-2cb4-4d1f-ba67-671f76c19537" />
> 
> I have a unit tests failing, so I will be grateful if you add unit tests for the missing functions and fix the failing unit tests.
> Dou you want to be assigned on this @Sandipmandal25 ?

Yes please assign this i will took a look into it from my side cc @MVrachev

**Comment by MVrachev** [comment]
You are assigned.

**Comment by Sandipmandal25** [comment]
hey @MVrachev  so i was trying to reproduce the failing test and noticed  that it doesnt always fail most runs pass but running the full test suite in parallel a few times it shows up also i got something diff  `AttributeError: 'stub' object has no attribute 'signed'` (you got str) so the exact error changes depending on which worker runs which test together. maybe the issue is that some tests patch things directly without proper cleanup so the changes leak into other tests in the same worker. there are more than 10 what i observed till now of these in `test_repository.py` so i am thinking to  is to convert all of them to use `monkeypatch.setattr`. does that sound right?
<img width="917" height="213" alt="Image" src="https://github.com/user-attachments/assets/5d923d1b-5f9e-490e-9ea4-5ba868a1bf6c" />

**Comment by Sandipmandal25** [comment]
also @MVrachev is this approch looks good? i have done the chnages in local waiting for your approval before making the pr! Thanks!
